utils/cla.ts

```diff
@@ -0,0 +1,208 @@
+			client_email:                process.env.GCLOUD_CLIENT_EMAIL,
+			client_id:                   process.env.GCLOUD_CLIENT_ID,
+			auth_uri:                    'https://accounts.google.com/o/oauth2/auth',
+			token_uri:                   'https://oauth2.googleapis.com/token',
+			auth_provider_x509_cert_url: 'https://www.googleapis.com/oauth2/v1/certs',
+			client_x509_cert_url:        process.env.GCLOUD_CLIENT_X509_CERT_URL
+		} as any
+	});
+	return storage.bucket(process.env.GCLOUD_BUCKET!);
+}
+
+export async function getPDFDownloadLink(username: string): Promise<string> {
+	username = username.trim().toLowerCase();
+	const bucket = getCLABucket();
+	const pdfFile = bucket.file(`${username[0]}/${username}.pdf`);
+	const [pdfURL] = await pdfFile.getSignedUrl({
+		action: 'read',
+		expires: Date.now() + 2 * 60 * 1000
+	});
+	return pdfURL;
+}
+
+export async function checkCLA(username: string): Promise<CLAInfo | null> {
+	username = username.trim().toLowerCase();
+
+	if (cache[username] && cache[username].ts < (Date.now() + 5 * 60 * 1000)) {
+		return cache[username].info;
+	}
+
+	try {
+		const bucket = getCLABucket();
+		const jsonFile = bucket.file(`${username[0]}/${username}.json`);
+		const [data] = await jsonFile.download();
+		const info: CLAInfo = JSON.parse(data.toString());
+
+		cache[username] = {
+			info,
+			ts: Date.now()
+		}
+
+		return info;
+	} catch {
+		return null;
+	}
+}
+
+export interface CreatePDFData {
+	fullname: string;
+	title: string;
+	company: string;
+	email: string;
+	signatureFile: string;
+	githubUsername: string;
+	today: string;
+};
+
+export async function createPDF({
+	fullname,
+	title,
+	company,
+	email,
+	signatureFile,
+	githubUsername,
+	today
+}: CreatePDFData): Promise<string> {
+	const pdfBytes = await readFile(join('public', CONTRIBUTOR_PDF));
+	const pdfDoc = await PDFDocument.load(pdfBytes);
+	const font = await pdfDoc.embedFont(StandardFonts.CourierBold);
+	const pages = pdfDoc.getPages();
+	const page = pages[4];
+	const { height: pageHeight } = page.getSize();
+
+	const drawText = (x: number, y: number, str: string) => {
+		page.drawText(str, {
+			x,
+			y: pageHeight - y,
+			size: 10,
+			font
+		});
+	};
+
+	const drawImage = async (x: number, y: number, file: string) => {
+		const bytes = await readFile(file);
+		const image = await pdfDoc.embedPng(bytes);
+		const imageRatio = image.width / image.height;
+		const bounds = { width: 210, height: 20 };
+		const boundsRatio = bounds.width / bounds.height;
+		let w, h;
+
+		if (imageRatio > boundsRatio) {
+			w = bounds.width;
+			h = image.height * (bounds.width / image.width);
+		} else {
+			w = image.width * (bounds.height / image.height);
+			h = bounds.height;
+		}
+
+		page.drawImage(image, {
+			x,
+			y: pageHeight - y,
+			width: w,
+			height: h
+		});
+	};
+
+	// page.drawRectangle({
+	// 	x: 80,
+	// 	y: pageHeight - 188,
+	// 	width: 210,
+	// 	height: 24,
+	// 	color: rgb(1, 0, 0)
+	// });
```

Comment:
```suggestion
```

remove

Reply:
These serve a purpose. I added a comment to explain their use.